### PR TITLE
Add asm test for arm32 il lifter

### PR DESCRIPTION
**Pull request by Heersin**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

current problem: some instructions are disassembly as "invalid" with new capstone
marked them as broken test for now.
- `dB "vmov.f64 d0, 1.250000e+00" 040bb7ee`
- `dB "vmrs apsr_nzcv, fpscr" 10faf1ee`
- `dB "vadd.f64 d0, d1, d2" 020b31ee`
- `dB "vsub.f64 d0, d1, d2" 420b31ee`
- `dB "vabs.f64 d1, d8" c81bb0ee`

**Test plan**

**Closing issues**

### CODE REVIEW

File: librz/analysis/arch/arm/arm_il32.c [hunk 1]
```diff
@@ -2589,7 +2596,6 @@ static RzILOpEffect *write_reg_lane(arm_reg reg, ut32 lane, ut32 vec_size, RzILO
  */
 static RzILOpEffect *vmov(cs_insn *insn, bool is_thumb) {
 	if (OPCOUNT() < 2 || !ISREG(0) || (!ISIMM(1) && !ISREG(1))) {
-		// for now, only support vmov rt, rn
```
Comment (XVilka) [inline librz/analysis/arch/arm/arm_il32.c:2592]: Not anymore? But the code hasn't changed

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/analysis/arch/arm/arm_il32.c [hunk 2]
```diff
@@ -3021,7 +3028,7 @@ static RzILOpEffect *vldn_multiple_elem(cs_insn *insn, bool is_thumb) {
 	for (int i = 0; i < n_groups; ++i) {
 		for (int j = 0; j < lanes; ++j) {
 			RzILOpBitVector *data0, *data1, *data2, *data3;
-			ut32 vreg_idx = i * regs;
+			ut32 vreg_idx = i * n;
```
Comment (Rot127) [inline librz/analysis/arch/arm/arm_il32.c]: Please rename `n` to something more descriptive.

File: test/db/asm/arm_32 [hunk 3]
```diff
@@ -641,3 +641,163 @@ d "vpop {d2, d3}" 042bbdec 0x0 (seq (set base (var sp)) (set d2 (loadw 0 64 (+ (
...
... 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (cast 8 false (>> (var d1) (bv 8 0x38) false))) (storew 0 (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (var r0) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (cast 8 false (>> (var d2) (bv 8 0x38) false))) (storew 0 (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (+ (var r0) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (bv 32 0x1)) (cast 8 false (>> (var d3) (bv 8 0x38) false))) empty)
```
Comment (Rot127) [inline test/db/asm/arm_32:801]: Not specific to this PR, but is it possible to replace those `(+ (+ (+ (+ (+ (+ (+ (+ (+ (+` with a loop?

**Review by Rot127 — APPROVED** [review]
Besides the variable name `n` LGTM. I investigate the issue with the broken instructions now. They work again in `v6`. So hopefully nothing fundamental is broken.

File: librz/analysis/arch/arm/arm_il32.c (hunk 2 above)
Comment (Rot127) [reply librz/analysis/arch/arm/arm_il32.c]: Also below.

**Comment by Rot127** [comment]
@Heersin The instructions you listed above are `arm v8` instructions. That is why Capstone (correctly) doesn't disassemble them. The `v8` feature flag must be set for them.

CS

without `v8`

```
> ./cstool arm 0x04,0x0b,0xb7,0xee,0x10,0xfa,0xf1,0xee,0x02,0x0b,0x31,0xee,0x42,0x0b,0x31,0xee,0xc8,0x1b,0xb0,0xee
ERROR: invalid assembly code
```

with `v8`
```
> ./cstool armv8 0x04,0x0b,0xb7,0xee,0x10,0xfa,0xf1,0xee,0x02,0x0b,0x31,0xee,0x42,0x0b,0x31,0xee,0xc8,0x1b,0xb0,0xee
 0  04 0b b7 ee  vmov.f64	d0, #1.250000e+00
 4  10 fa f1 ee  vmrs	apsr_nzcv, fpscr
 8  02 0b 31 ee  vadd.f64	d0, d1, d2
 c  42 0b 31 ee  vsub.f64	d0, d1, d2
10  c8 1b b0 ee  vabs.f64	d1, d8
```

For proof the LLVM check:

Without `v8`
```
echo "0x04,0x0b,0xb7,0xee,0x10,0xfa,0xf1,0xee,0x02,0x0b,0x31,0xee,0x42,0x0b,0x31,0xee,0xc8,0x1b,0xb0,0xee" | llvm-mc-16 --triple=arm --disassemble
	.text
<stdin>:1:1: warning: invalid instruction encoding
0x04,0x0b,0xb7,0xee,0x10,0xfa,0xf1,0xee,0x02,0x0b,0x31,0xee,0x42,0x0b,0x31,0xee,0xc8,0x1b,0xb0,0xee
^
<stdin>:1:21: warning: invalid instruction encoding
0x04,0x0b,0xb7,0xee,0x10,0xfa,0xf1,0xee,0x02,0x0b,0x31,0xee,0x42,0x0b,0x31,0xee,0xc8,0x1b,0xb0,0xee
                    ^
<stdin>:1:41: warning: invalid instruction encoding
0x04,0x0b,0xb7,0xee,0x10,0xfa,0xf1,0xee,0x02,0x0b,0x31,0xee,0x42,0x0b,0x31,0xee,0xc8,0x1b,0xb0,0xee
                                        ^
<stdin>:1:61: warning: invalid instruction encoding
0x04,0x0b,0xb7,0xee,0x10,0xfa,0xf1,0xee,0x02,0x0b,0x31,0xee,0x42,0x0b,0x31,0xee,0xc8,0x1b,0xb0,0xee
                                                            ^
<stdin>:1:81: warning: invalid instruction encoding
0x04,0x0b,0xb7,0xee,0x10,0xfa,0xf1,0xee,0x02,0x0b,0x31,0xee,0x42,0x0b,0x31,0xee,0xc8,0x1b,0xb0,0xee
                                                                                ^
```

with `v8`
```
echo "0x04,0x0b,0xb7,0xee,0x10,0xfa,0xf1,0xee,0x02,0x0b,0x31,0xee,0x42,0x0b,0x31,0xee,0xc8,0x1b,0xb0,0xee" | llvm-mc-16 --triple=armv8 --disassemble
	.text
	vmov.f64	d0, #1.250000e+00
	vmrs	APSR_nzcv, fpscr
	vadd.f64	d0, d1, d2
	vsub.f64	d0, d1, d2
	vabs.f64	d1, d8
```

**Comment by Rot127** [comment]
@Heersin Sorry. The stuff above is not correct. `llvm-mc` does indeed disassembles those instructions with `--mattr=vfp3`. Will need to fix it in CS.

**Comment by Rot127** [comment]
https://github.com/capstone-engine/capstone/pull/2090

**Comment by Heersin** [comment]
> [capstone-engine/capstone#2090](https://github.com/capstone-engine/capstone/pull/2090)

ok is there anything i can do ？

File: librz/analysis/arch/arm/arm_il32.c (hunk 2 above)
Comment (Heersin) [reply librz/analysis/arch/arm/arm_il32.c]: oh it's related to the "n" in "vld**n**". I will rename it to a more specific variable name.

File: test/db/asm/arm_32 (hunk 3 above)
Comment (Heersin) [reply test/db/asm/arm_32:801]: I will check if `repeat` could be used here

File: librz/analysis/arch/arm/arm_il32.c (hunk 1 above)
Comment (Heersin) [reply librz/analysis/arch/arm/arm_il32.c:2592]: yep, it's a comment left by florian when create arm32 lifter. can be removed now.

**Comment by XVilka** [comment]
> > [capstone-engine/capstone#2090](https://github.com/capstone-engine/capstone/pull/2090)
> 
> ok is there anything i can do ？

Nothing for now. Ideally, it would be nice to have a patch release with some fixes: https://github.com/capstone-engine/capstone/issues/2081
But that's up to Capstone maintainers, not us. Sadly, we have no power there.

**Comment by XVilka** [comment]
@Heersin so it is probably not a draft anymore? Lets merge and check the tests again once capstone 5.0.1 or 5.1 released

**Comment by Heersin** [comment]
> @Heersin so it is probably not a draft anymore? Lets merge and check the tests again once capstone 5.0.1 or 5.1 released

ok, after variable rename I will mark it as ready for review

**Review by XVilka — APPROVED** [review]

**Pull request merged**